### PR TITLE
Add first draft of contribution guides

### PR DESCRIPTION
**Pull request by leoswaldo**
This first commit contains a draft for the contribution
guidelines of the project, it is subject to be updated
continuously as the project matures.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+1. The solution most be related to an issue in this repository
+2. The solution most be created in a separate repository
+3. The solution most contain valid documentation (to-be-defined)
+4. The solution most solve a generic issue
+
+If you already have a solution and want to share it through this channel please contact the project owners (mail-to-be-added-here)
+
+
+## Commit Format
```
Comment (charlycastro) [inline CONTRIBUTING.md]: should we add the .gitmessage template in this repo later?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Contributing to From Zero To Hero
+
+From Zero To Hero is an open source project
+
+# Creating a New Solution
+
+A new solution must be related to an issue in this repository, if you have a new solution proposal or would like a project to be developed by the community please contact FIRST the project owners (mail-to-be-added-here) to discuss the scope and opportunities. Later some important aspects would need to be met as listed below
```
Comment (charlycastro) [inline CONTRIBUTING.md]: change to active voice vs passive voice 

```
 if you have a new solution proposal or would like the community to develop a project ```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+1. The solution most be related to an issue in this repository
+2. The solution most be created in a separate repository
+3. The solution most contain valid documentation (to-be-defined)
```
Comment (charlycastro) [inline CONTRIBUTING.md]: most -> must

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+4. The solution most solve a generic issue
+
+If you already have a solution and want to share it through this channel please contact the project owners (mail-to-be-added-here)
```
Comment (charlycastro) [inline CONTRIBUTING.md]: adding comma
```If you already have a solution and want to share it through this channel, please contact the project owners (mail-to-be-added-here)```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+## Commit Format
+
+### General Rules
+
+Always refer as third person Use English
+
+#### Components
+- Subject (Required )
+- Description (Not always required when subject is enough)
```
Comment (charlycastro) [inline CONTRIBUTING.md]: ```- Description (Not always required when the subject is enough)````

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+- Signed (Required)
+
+#### Subject
+
+- Limit line length up to 50 characters
+- Capitalize only firste letter in the subject unless a name is used in the rest of the subject
```
Comment (charlycastro) [inline CONTRIBUTING.md]: firste >  first

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+- Do not end line with dot
```
Comment (charlycastro) [inline CONTRIBUTING.md]: with a dot?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+- First word should always be a verb for explanatory (Imperative)
```
Comment (charlycastro) [inline CONTRIBUTING.md]: consider adding 
```The first word should always ....```

Comment (leoswaldo) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 7 above)
Comment (leoswaldo) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 6 above)
Comment (leoswaldo) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 5 above)
Comment (leoswaldo) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 4 above)
Comment (leoswaldo) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 3 above)
Comment (leoswaldo) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 2 above)
Comment (leoswaldo) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 1 above)
Comment (leoswaldo) [reply CONTRIBUTING.md]: yeah, tha'll be cool, I think once we start having our first issues and see the common stuff we should add it

**Pull request merged**